crates/datasources/src/bson/errors.rs

```diff
@@ -35,6 +41,9 @@ pub enum BsonError {
         datafusion::arrow::datatypes::DataType,
     ),
 
+    #[error("value was type {0}, expected {1}")]
+    IncorrectType(DataType, DataType),
```

Comment:
Personally I'd make this a struct like this:

```suggestion
    #[error("value was type {got}, expected {expected}")]
    IncorrectType{got: DataType, expected: DataType},
```

Makes it unambiguous which field is which.

Reply:
This does seem like a better solution, and I think I did this because I'm a `n00b`.

This duplicates an error in the `datasources::bson::errors::BsonError` enum, and so I'm inclined to come back and change it later when/where I can do both in a very boring PR.